flink-formats/flink-json/src/test/java/org/apache/flink/formats/json/JsonRowDataSerDeSchemaTest.java

```diff
@@ -375,8 +375,12 @@ private void testParseErrors(TestSpec spec) throws Exception {
 		JsonRowDataDeserializationSchema failingSchema = new JsonRowDataDeserializationSchema(
 			spec.rowType,  new RowDataTypeInfo(spec.rowType), false, false);
 
-		thrown.expectMessage(spec.errorMessage);
-		failingSchema.deserialize(spec.json.getBytes());
+		try {
+			failingSchema.deserialize(spec.json.getBytes());
+			throw new RuntimeException("expecting exception " + spec.errorMessage);
+		} catch (Throwable t) {
+			assertEquals(t.getMessage(), spec.errorMessage);
+		}
```

Comment:
IMO, `thrown.expectMessage` is more concise for the exception validation. Do we have a reason to change this?

Reply:
It's a very interesting question. The original implementation indeed has some unexpected behavior. It will abort the testing when it encounters one exception, no matter it is expected or not.
Hence, the original tests only run until the first exception it thrown, and the next tests will be omitted. That's why the next test can pass before.

Reply:
Thanks for the finding, I didn't know that before. Then I think we should replace the other usage of `thrown` in this test.

Reply:
Thanks for the reminder, I've corrected all other misusage in this test.